src/objectid.ts

```diff
@@ -37,8 +61,26 @@ export class ObjectId extends BSONValue {
 
   static cacheHexString: boolean;
 
-  /** ObjectId Bytes @internal */
-  private buffer!: Uint8Array;
+  /**
+   * The size of the current ObjectId buffer pool.
+   */
+  static get poolSize(): number {
+    return poolSize;
+  }
+
+  static set poolSize(size: number) {
+    const iSize = Math.ceil(size); // Ensure new pool size is an integer
+    if (iSize <= 0) {
+      throw new BSONError('poolSize must be a positive integer greater than 0');
+    }
+    poolSize = iSize;
+  }
```

Comment:
```suggestion
  static set poolSize(size: number) {
    poolSize = Math.max(Number(size) >>> 0, 1);
  }
```

I like where this is going but `ceil` isn't actually going to make sure this value is an integer, `NaN` and `Infinity` are going to sneak by the following condition. Rather than adding error handling for those cases, I think this setter should just be an "always works" kind of API, so given any input it makes a number out of it, and that number must be positive and probably no greater than UInt32.max (which would be 51GB of ObjectIds 😵‍💫). Error checking could be done externally by reading the value after it has been written and checking for equality.

Reply:
I like it! My only thought here is passing in a negative number results in a huge poolSize. Which could silently lead to a pretty significant memory leak if a user accidentally passed in `-1` for example. 

`poolSize = Math.max(Number(-1) >>> 0, 1); // 4294967295`

I think adding in a `Math.abs` makes it a bit safer. WDYT?

`poolSize = Math.max(Math.abs(Number(-1)) >>> 0, 1); // 1`

Reply:
Good catch & great idea! 🚀